wagtail_live/models.py

```diff
@@ -59,6 +58,168 @@ class LivePageMixin(models.Model):
         StreamFieldPanel("live_posts"),
     ]
 
+    def _get_live_post_index(self, live_post_id, low, high):
+        """ Retrieve index of a live post in page's live posts.
+
+        Uses binary search since live posts are sorted by timestamp.
+        This implementation may change.
+
+        Args:
+            live_post_id (str):
+                ID of the live post to look for.
+            low (int):
+                Index of first element in page live posts.
+            high (int):
+                Index of last element in page live_posts.
+
+        Returns:
+            (int) Index of the live post if found else -1
+        """
+        
+        while low < high:
+            mid = (low + high) // 2
+            curr_id = self.live_posts[mid].id
+
+            if float(curr_id) == float(live_post_id):
+                return mid
+
+            elif float(curr_id) < float(live_post_id):
+                low = mid + 1
+
+            else:
+                high = mid - 1
+        
+        if self.live_posts[low].id != live_post_id:
+            return -1
+        return low
+
+    def get_live_post_index(self, live_post_id):
+        """Retrieve index of a livepost with its ID.
+        
+        Args:
+            live_post_id (str); Id of the live post to look for.
+
+        Returns:
+            (int) Index of the live post if found else -1
+        """
+
+        return self._get_live_post_index(live_post_id, 0, len(self.live_posts))
+
+    def get_live_post_by_index(self, live_post_index):
+        """Retrieve a live post by its index.
+        
+        Args: 
+            live_post_index (str): Index of the live post to look for.
+
+        Returns:
+            (LivePostBlock) The live post instance
+
+        Raises:
+            (someException) if a live post with the given index doesn't exist.
+        """
+
+        try:
+            live_post = self.live_posts[live_post_index]
+        except KeyError:
+            raise 
+        return live_post
+
+    def get_live_post_by_id(self, live_post_id):
+        """Retrieve a live post by its ID.
+        
+        Args: 
+            live_post_id (str): ID of the live post to look for.
+
+        Returns:
+            (LivePostBlock) The live post instance
+
+        Raises:
+            (someException) if a live post with the given ID doesn't exist.
+        """
+
+        live_post_index = self.get_live_post_index(live_post_id=live_post_id)
+        if live_post_index == -1:
+            raise
```

Comment:
This can never be reached I think, because self.get_live_post_index will already throw an error if something goes wrong, instead of returning -1.